### PR TITLE
pre-commit: add markdown lint

### PR DESCRIPTION
**Pull request by macpijan**
Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

### CODE REVIEW

**Comment by macpijan** [comment]
There are **many** more fixes to go, which cannot be fixed automatically. They can be listed with `pre-commit run --all-files`

**Comment by macpijan** [comment]
I wanted to run it in docker, so there is no need to pollute host PC with nodejs runtime (https://github.com/Dasharo/docs/pull/85/files#diff-63a9c44a44acf85fea213a857769990937107cf072831e1a26808cfde9d096b9R26)

Unfrotunately, precommit-ci does not support docker it seems: https://results.pre-commit.ci/run/github/316342881/1646260993.OfRbnUS1Q-GuArbT7Pk_jQ

File: docs/variants/asus_kgpe_d16/spi-wp.md [hunk 1]
```diff
@@ -152,46 +152,46 @@ the chip (running from the tagret machine to be protected):
...
+ start=0x00020000 length=0x007e0000 (upper 63/64)
+ start=0x00000000 length=0x007f8000 (lower 255/256)
+ start=0x00008000 length=0x007f8000 (upper 255/256)
+ start=0x00000000 length=0x007fc000 (lower 511/512)
+ start=0x00004000 length=0x007fc000 (upper 511/512)
+ start=0x00000000 length=0x007fe000 (lower 1023/1024)
+ start=0x00002000 length=0x007fe000 (upper 1023/1024)
+ start=0x00000000 length=0x007ff000 (lower 2047/2048)
+ start=0x00001000 length=0x007ff000 (upper 2047/2048)
+ start=0x00000000 length=0x00800000 (all)
```
Comment (pietrushnic) [inline docs/variants/asus_kgpe_d16/spi-wp.md:194]: @macpijan I'm not sure about such changes, can we disable automatic fixes for code blocks?

File: docs/variants/clevo_nv41/fwupd_usage.md [hunk 2]
```diff
@@ -159,21 +170,23 @@ An update requires the system to shutdown to complete. Shutdown now? [y|N]:
...
-6.	c6a80ac3a22083423992a3cb15018989f37834d6 (TPM)
-7.	eefcbd318bd31fc1eba6358e628b3f9dceb87206 (USB4 host controller)
+0. Cancel
+1. 4bde70ba4e39b28f9eab1628f9dd6e6244c03027 (11th Gen Intel Core™ i7-1165G7 @ 2.80GHz)
+2. 5792b48846ce271fab11c4a545f7a3df0d36e00a (Display controller)
+3. 073c01931cb0e9889bbfb2ea4a4c2fc558805fc6 (Display controller)
+4. dbee8bd3b1ae0316ad143336155651eedb495a0e (NV4XMB,ME,MZ)
+5. 71b677ca0f1bc2c5b804fa1d59e52064ce589293 (SSD 980 PRO 1TB)
+6. c6a80ac3a22083423992a3cb15018989f37834d6 (TPM)
+7. eefcbd318bd31fc1eba6358e628b3f9dceb87206 (USB4 host controller)
```
Comment (pietrushnic) [inline docs/variants/clevo_nv41/fwupd_usage.md:185]: same here, this is output pasted from console, maybe visually-wise it doesn't make difference

**Review by pietrushnic — COMMENTED** [review]
I like this pre-commit checks, it would improve quality of documentation, I think I have some rules to be added like the way we write open-source firmware or open-source in different context. I'm not sure how much tool can do.

File: docs/variants/asus_kgpe_d16/spi-wp.md (hunk 1 above)
Comment (macpijan) [reply docs/variants/asus_kgpe_d16/spi-wp.md:194]: @pietrushnic Yes, actually I have already disabled it here: https://github.com/Dasharo/docs/pull/85/files#diff-39aecbc26791dd03b9a4fdc1f856769ea73308294b7111d70f279ac9bdef35bdR53

Just need to rerun and force push

**Comment by macpijan** [comment]
> I think I have some rules to be added like the way we write open-source firmware or open-source in different context. I'm not sure how much tool can do.

It can do **a lot**. This sounds definitely feasible. List of hooks is there: https://pre-commit.com/hooks.html
`consistent-spelling` might be useful here. We can easily add our own scripts and different tools as custom hooks as well.

File: docs/variants/asus_kgpe_d16/spi-wp.md (hunk 1 above)
Comment (macpijan) [reply docs/variants/asus_kgpe_d16/spi-wp.md]: disabled for code blocks

File: docs/variants/clevo_nv41/fwupd_usage.md (hunk 2 above)
Comment (macpijan) [reply docs/variants/clevo_nv41/fwupd_usage.md]: disabled for code blocks

**Comment by macpijan** [comment]
This looks interesting for the next hook to enforce our commiting rules: https://github.com/talos-systems/conform

**Review by pietrushnic — APPROVED** [review]

**Comment by macpijan** [comment]
@pietrushnic Yes. the problem is the markdown linter is in NodeJS. As explained here: https://github.com/Dasharo/docs/pull/85#issuecomment-1057471445

We have two options:
1. keep using the markdown linter in container, but pre-commit CI will not work - we can disable it or replace by GitHub actions later
2. use markdown linter natively, which would require all users of this repo to install node runtime on the host (or prepare otherwise compatible environment)

**Comment by pietrushnic** [comment]
Understood, do we want to fix that as part of this PR or other?

**Pull request merged**